### PR TITLE
Move `Type` to `core` package

### PR DESCRIPTION
**Pull request by Cryrivers**

### CODE REVIEW

File: packages/@manta-style/core/src/index.ts [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
 export type MantaStyleContext = {
   query: { [key: string]: unknown };
   plugins: PluginSystem;
 };
 export * from './plugin';
-export { annotationUtils };
+export { annotationUtils, Annotation };
+
+export abstract class Type {
+  abstract deriveLiteral(
```
Comment (tanhauhau) [inline packages/@manta-style/core/src/index.ts]: move type definition into `type` folder? 
so the `index.ts` is just import and export?

**Review by tanhauhau — COMMENTED** [review]
LGTM

File: packages/@manta-style/core/src/index.ts (hunk 1 above)
Comment (Cryrivers) [reply packages/@manta-style/core/src/index.ts]: eventually we will do so. due to TypeScript bug, we are unable to have a file containing only types.

Comment (tanhauhau) [reply packages/@manta-style/core/src/index.ts]: oops. okay then

Comment (Cryrivers) [reply packages/@manta-style/core/src/index.ts:13]: it's actually a bug in `awesome-typescript-loader`. maybe we could arrange some time to do this refactor.

**Pull request merged**